packages/atlas-service/src/make-automation-agent-op-request.ts

```diff
@@ -0,0 +1,135 @@
+type ClusterOrServerlessId =
+  | { serverlessId?: never; clusterId: string }
+  | { serverlessId: string; clusterId?: never };
+
+export type AutomationAgentRequestOpTypes = {
+  listIndexStats: [
+    ClusterOrServerlessId & { db: string; collection: string },
+    {
+      collName: string;
+      dbName: string;
+      indexName: string;
+      indexProperties: { label: string; properties: Record<string, unknown> }[];
+      indexType: { label: string };
+      keys: { name: string; value: string | number };
+      sizeBytes: number;
+      status: 'rolling build' | 'building' | 'exists';
```

Comment:
Pretty confident about those, but I'll do another pass through mms codebase to check whether or not I'm getting the types correctly here